packages/esm-ohri-pmtct-app/src/views/mch-summary/tabs/current-pregnancy.component.tsx

```diff
@@ -49,21 +49,36 @@ const CurrentPregnancy: React.FC<PatientChartProps> = ({ patientUuid }) => {
   const [infantOutcomes, setInfantOutcomes] = useState([]);
   const { formNames, encounterTypes, obsConcepts, formUuids } = useConfig();
   const [totalAncCount, setTotalAncCount] = useState(null);
+  const [motherStatus, setMotherStatus] = useState(null);
+  const [deliveryDate, setDeliveryDate] = useState(null);
+  const [motherHivStatus, setMotherHivStatus] = useState(null);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    const fetchData = async () => {
+    const fetchDataAndSetState = async () => {
       try {
-        const [totalAncCount] = await Promise.all([fetchMambaReportData('no_of_anc_visits')]);
+        setLoading(true);
+        const [totalAncCount, motherStatus, deliveryDate, motherHivStatus] = await Promise.all([
```

Comment:
> [!NOTE]
> My general thoughts

Could we have these as individual hook calls and then revamp the `fetchData` function to use `SWR` for data fetching: 
+ This would eliminate the need to have multiple states for each bit of information that you are fetching.
+ The each data fetch would be independent and not rely on the entire `Promise` to resolve.
+ `SWR` would automatically handle things such as loading states and error states which we are doing manually.